firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1136,6 +1136,10 @@ class BrowserViewController: UIViewController,
         if UIDevice.current.userInterfaceIdiom == .pad {
             topTabsViewController?.refreshTabs()
         }
+
+        if !focusUrlBar, urlBar.inOverlayMode {
+            urlBar.leaveOverlayMode(reason: .finished, shouldCancelLoading: false)
+        }
```

Comment:
With the upcoming web engine changes, could you add a comment on top of this saying this was for a specific bugfix? This will make it easier to track what patches landed to fix some url bar issue.

Reply:
Also, I'm not sure how you got in the state from the video. If I copy paste a URL on my device, the URL doesn't seem to be affected. Did you follow extra steps?

Reply:
hey @lmarceau sure I can add a comment. It was also strange for me to see that there was already a `focusUrlBar` property but not used. 

As for the steps to reproduce it, the url has to be in copy mode or as we call it overlay mode but has nothing to do with the actual copy. So what you would to is to tap the url to bring it in the overlay mode, swipe on home to dismiss the keyboard but still keep the url selected, and then navigate back and forth in between tab and see that the selection is kept even tho the url changes.